File: en_python/exo2-5.py
import os
import time
import random

start = time.time()  

pid1 = os.fork()
if pid1 == 0:
    t = random.randint(1, 10)
    print(f"fils 1 : je vais dormir {t} secondes")
    time.sleep(t)
    print("fils 1 terminé")
    exit(0)
else:
    pid2 = os.fork()
    if pid2 == 0:
        t = random.randint(1, 10)
        print(f"fils 2 : je vais dormir {t} secondes")
        time.sleep(t)
        print("fils 2 terminé")
        exit(0)
    else:
        os.wait()  
        os.wait() 
        end = time.time()
        print(f"Durée totale : {int(end - start)} secondes")

File: en_C/exo2-5.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>
#include <time.h>

int main() {
    pid_t pid1, pid2;
    time_t start, end;

    srand(time(NULL)); 
    start = time(NULL); 

    pid1 = fork();
    if (pid1 == 0) {
        
        int t = rand() % 10 + 1;
        printf("fils 1 : je vais dormir %d secondes\n", t);
        sleep(t);
        printf("fils 1 terminé\n");
        exit(0);
    } else {
        pid2 = fork();
        if (pid2 == 0) {
        
            int t = rand() % 10 + 1;
            printf("fils 2 : je vais dormir %d secondes\n", t);
            sleep(t);
            printf("fils 2 terminé\n");
            exit(0);
        } else {
            
            wait(NULL);
            wait(NULL);
            end = time(NULL);
            printf("Durée totale : %ld secondes\n", end - start);
        }
    }

    return 0;
}
